invenio_records_resources/services/records/service.py

```diff
@@ -34,13 +34,20 @@ class RecordService(Service):
     def indexer(self):
         """Factory for creating an indexer instance."""
         return self.config.indexer_cls(
+            # the routing key is mandatory in the indexer constructor since
```

Comment:
important explanation of why the duplicated parameter passing (queue name as routing key)